app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/OtlpTelemetry.java

```
@@ -72,11 +75,26 @@ public Iterable<String> keys(ExtractModel carrier) {
         return W3CTraceContextPropagator.getInstance().extract(Context.current(), model, getter);
     }
 
-    private Span startOTLPSpan(String spanName, Context context) {
-        return tracer.spanBuilder(spanName)
-                .setSpanKind(SpanKind.SERVER)
-                .setParent(context)
-                .startSpan();
+    public Span startOTLPSpan(String spanName, Context context, Span parentSpan) {
+        if (tracer == null) {
+            return null;
+        }
+
+        if (isBlank(spanName)) {
+            throw new AppsmithException(AppsmithError.INVALID_PARAMETER, "spanName");
+        }
```

We can return null over here ...a blank span name can be ignore.

---

Hi @vsvamsi1 Sharing my perspective on this: I think we should return an error here because span name being blank is a developer error ie. it is only possible if a developer has made a mistake when creating a span. cc: @rajatagrawal